opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -99,23 +167,21 @@ where
         event: &tracing::Event<'_>,
         _ctx: tracing_subscriber::layer::Context<'_, S>,
     ) {
-        let meta = event.metadata();
+        let normalized_meta = event.normalized_metadata();
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
         let mut log_record: LogRecord = LogRecord::default();
         log_record.severity_number = Some(severity_of_level(meta.level()));
         log_record.severity_text = Some(meta.level().to_string().into());
 
-        // add the `name` metadata to attributes
-        // TBD - Propose this to be part of log_record metadata.
-        let vec = vec![("name", meta.name())];
-        log_record.attributes = Some(vec.into_iter().map(|(k, v)| (k.into(), v.into())).collect());
-
         // Not populating ObservedTimestamp, instead relying on OpenTelemetry
         // API to populate it with current time.
 
-        let mut visitor = EventVisitor {
-            log_record: &mut log_record,
-        };
+        let mut visitor = EventVisitor::default();
+        visitor.visit_metadata(meta);
```

Comment:
I am not very convinced with visiting metadata, as it seems all the metadata fields it is processing are not part of the semantic convention (including `log.name`). If that is the case, should we move this `visit_metadata` call under the experimental flag?

Reply:
I renamed log.name just name like before. All the other fields in visit_metadata are under the experimental flag. Is it ok?